Add PBC database as demostration data

DougManuel opened the pull request:
This pull request adds the PBC data to demonstrate how to create and use the package. The PBC data is openly accessible and available in the `survival` package. 
Included are:
- the data: pbc.Rd
- variables sheet: pbc_variables.csv
- variable details sheet: pbc_variable_details.csv
- dataset metadata: pbc_database.yaml. This is new file, and it may require a discussion. At this point, it is a simple meta to describe the data provenance, but this file could potentially be used for a different purposes, such as generating a data dictionary. For now, it is simply introduced as a good data management practice. 

note: mtcars is also included in the package, and this data requires similar improvements and additions as is proposed for the pbc data.

reikookamoto commented on inst/extdata/pbc.Rd:
I noticed that this file don't appear to be generated from a `roxygen2` docstring defined in a .R file. Could you please clarify where it came from?

Also, I want to point out that `.Rd` files should be stored in the `man/` directory. If an `.Rd` file isn't located there, I think it can lead to issues like documentation accessibility (i.e., users won't be able to access the documentation using `?data` or `help(data)`.

DougManuel replied:
This file is currently for reference only. It comes from the `survival` library. It should be removed when we finish the data refactoring. 

However, there is an interesting challenge. 'data. R uses roxygen2 to generate documentation for data. Great! BUT, there needs to be a better workflow for reusing the roxygen2 data as metadata for other purposes. A major issue is quite an unstructured approach to using oxygen for data documentation in R. For example, note the `\tab` and `\cr` characters in PBC.Rd. This formatting isn't standard and would need to be parsed for the labels to be reused in R for tables and figures. 

From the original `pbc.Rd` file for documenting the `pbc` data.
```
\format{
  \tabular{ll}{
    age:\tab in years\cr
    albumin:\tab serum albumin (g/dl)\cr
    alk.phos:\tab alkaline phosphotase (U/liter)\cr
 ```

Essentially in `recodeflow` what we've done with `variable.csv` and `variable_details` is generate a more machine-actionable version of the traditional roxygen with additional metadata. We started with metadata for recoding variables, but this metadata can now be reused -- unlike pbc.Rd.

I've carried this thinking through to the main descriptive metadata for pbc.Rd by creating a new YAML that follows the Dublin Core metadata approach. This needs to be talked about... 

Although it may not be commonly used, having reusable metadata for a dataset within an R workflow seems important. One use is creating a data dictionary for your study. 

However, and the .Rd files don't work well for keeping your metadata alive. 

A better approach, it seems to me, is to use a standard metadata like Dublin in YAML format, _then_ generate the .Rd file from the Dublin YAML for generating documentation for CRAN and pkgdown.

DougManuel replied:
pbc.Rd deleted. 

The information from pbc.Rd is in:
- data.R
- pbc_metadata.yaml

See https://github.com/Big-Life-Lab/recodeflow/issues/46

reikookamoto commented on R/pbc_worksheets.R:
Code that generates or processes example datasets (e.g., scripts that prepare data and end with a call to use_data()) shouldn't be placed in the R/ directory. For guidance, see [7.1 Exported data](https://r-pkgs.org/data.html#sec-data-data) of the R Packages book.

To preserve the "origin story" of your data, it’s recommended to store data preparation scripts in the data-raw/ directory. This keeps the data preparation steps separate from the package's main code, as discussed in [7.1.1 Preserve the origin story of package data](https://r-pkgs.org/data.html#sec-data-data-raw).

DougManuel replied:
prep_pbc_data.R now imports all data. It is stored on data-raw.

reikookamoto reviewed (commented):
Review the files to ensure they align with the directory structure and organization best practices outlined in the *R Packages* book

DougManuel commented:
- Documentation website not yet building on my computer with errors at the end of derived_variables.Rmd. 

Bug fixes and improvements in recent commits include:
1. Renamed data files for a more consistent approach to address bugs and confusion. See #details in data.R. 

**pbc.rda** - Primary Biliary Cirrhosis (PBC) data.
**pbc_metadata.rda** - Metadata for the PBC data. A list containing the database name and other identification information.
**pbc_variables.rda** - `variables` metadata for the PBC dataset. Metadata for each variable in the PBC data. A data frame with 24 rows and 11 columns.
**pbc_variable_details.rda** - `variable_details` metadata for the PBC dataset. Metadata for variable category or value label in the PBC data. https://github.com/Big-Life-Lab/recodeflow/blob/dcf600e15762c59d6a28b9959b13f8130b5a18ea/R/data.R#L7

2. Updated pbc_variables:
- Address bugs in NA data, incorrect names and labels, the addition of the age group categorical variables, and derived variable example 'example_der`.

3. Changed data in derived_variables.Rmd to pbc data. 
- bugs remain in `derived1` example. Likely from an error in `pbc_variable_details.rda`, but an error is recode_with_table() occurs. 
https://github.com/Big-Life-Lab/recodeflow/blob/dcf600e15762c59d6a28b9959b13f8130b5a18ea/vignettes/derived_variables.Rmd#L83
 
4. `prep_pbc_data.R` updated to generate all .rda data. Remember to execute this file when you change `pbc_variables.csv` or the other 'raw' data.

DougManuel commented on R/data.R:
```diff
@@ -0,0 +1,65 @@
+#' Worksheet for mtcars dataset
```
mtcars references are removed. We can start with pbc and add mtcars in the future.

DougManuel commented:
The package now passes checks and builds.

@reikookamoto @yulric Do you have any other comments to address? 

We now have the PBC data for vignettes and for users to test functions.

We'll need to refresh a few vignettes with minimal examples to use alongside the pbc data.
